substrate/frame/ranked-collective/src/benchmarking.rs

```diff
@@ -56,131 +58,263 @@ fn make_member<T: Config<I>, I: 'static>(rank: Rank) -> T::AccountId {
-		let r in 0 .. 10;
+	#[benchmark]
+	fn promote_member(r: Linear<0, 10>) -> Result<(), BenchmarkError> {
+		// Convert `r` to a rank and create the member.
 		let rank = r as u16;
 		let who = make_member::<T, I>(rank);
 		let who_lookup = T::Lookup::unlookup(who.clone());
+
+		// Try to fetch the promotion origin.
 		let origin =
 			T::PromoteOrigin::try_successful_origin().map_err(|_| BenchmarkError::Weightless)?;
-		let call = Call::<T, I>::promote_member { who: who_lookup };
-	}: { call.dispatch_bypass_filter(origin)? }
-	verify {
+
+		#[extrinsic_call]
+		_(origin as T::RuntimeOrigin, who_lookup);
+
+		// Ensure the member's rank has increased by 1.
 		assert_eq!(Members::<T, I>::get(&who).unwrap().rank, rank + 1);
+
+		// Ensure the correct event was emitted for the rank change.
 		assert_last_event::<T, I>(Event::RankChanged { who, rank: rank + 1 }.into());
+
+		Ok(())
 	}
 
-	demote_member {
-		let r in 0 .. 10;
+	#[benchmark]
+	fn demote_member(r: Linear<0, 10>) -> Result<(), BenchmarkError> {
+		// Convert `r` to a rank and create necessary members for the benchmark.
 		let rank = r as u16;
-		let first = make_member::<T, I>(rank);
 		let who = make_member::<T, I>(rank);
 		let who_lookup = T::Lookup::unlookup(who.clone());
 		let last = make_member::<T, I>(rank);
+
+		// Get the last index for the member.
 		let last_index = IdToIndex::<T, I>::get(rank, &last).unwrap();
+
+		// Try to fetch the demotion origin.
 		let origin =
 			T::DemoteOrigin::try_successful_origin().map_err(|_| BenchmarkError::Weightless)?;
-		let call = Call::<T, I>::demote_member { who: who_lookup };
-	}: { call.dispatch_bypass_filter(origin)? }
-	verify {
+
+		#[extrinsic_call]
+		_(origin as T::RuntimeOrigin, who_lookup);
+
+		// Ensure the member's rank has decreased by 1.
 		assert_eq!(Members::<T, I>::get(&who).map(|x| x.rank), rank.checked_sub(1));
-		assert_eq!(MemberCount::<T, I>::get(rank), 2);
+
+		// Ensure the member count remains as expected.
+		assert_eq!(MemberCount::<T, I>::get(rank), 1);
+
+		// Ensure the index of the last member has changed.
 		assert_ne!(last_index, IdToIndex::<T, I>::get(rank, &last).unwrap());
-		assert_last_event::<T, I>(match rank {
-			0 => Event::MemberRemoved { who, rank: 0 },
-			r => Event::RankChanged { who, rank: r - 1 },
-		}.into());
+
+		// Ensure the correct event was emitted depending on the member's rank.
+		assert_last_event::<T, I>(
+			match rank {
+				0 => Event::MemberRemoved { who, rank: 0 },
+				r => Event::RankChanged { who, rank: r - 1 },
+			}
+			.into(),
+		);
+
+		Ok(())
 	}
 
-	vote {
-		let class = T::Polls::classes().into_iter().next().unwrap();
-		let rank = T::MinRankOfClass::convert(class.clone());
+	#[benchmark]
+	fn vote() -> Result<(), BenchmarkError> {
+		// Find the first class or use a default one if none exists.
+		let (class_exists, class) = T::Polls::classes()
+			.into_iter()
+			.next()
+			.map(|c| (true, c))
+			.unwrap_or_else(|| (false, Default::default()));
```

Comment:
Also general if there doesn't exist any class, how can there exist a default value?

Reply:
would this also apply to default poll?

Reply:
`Default` on `Poll` type should be changed to `From<u32>`

Reply:
Maybe the best is to have some default rank and if the class doesn't exist the default rank is used. (Basically require Default for Rank). Then the rest should be solvable.

Reply:
We should consider `max_ongoing` returns the maximum amount of ongoing polls as `u32` in `runtime-benchmarks` feature flag, I think the polling `Index` should bound `From<u32>` by this logic? 

```rust
/// The maximum amount of ongoing polls within any single class. By default it practically
/// unlimited (`u32::max_value()`).
#[cfg(feature = "runtime-benchmarks")]
fn max_ongoing() -> (Self::Class, u32) {
	(Self::classes().into_iter().next().expect("Always one class"), u32::max_value())
}
```